2_process/src/combine_NLCD_PRMS.R

```diff
@@ -27,15 +27,17 @@ AOI_LC_w_area <- function(NLCD_LC_df, area_att, aoi_comids_df){
 ###----
 
 # Estimate prop of LC coverage at PRMS catchment scale
-proportion_lc_by_prms <- function(NLCD_LC_df_w_area, catchment_att = 'CAT'){
+proportion_lc_by_prms <- function(NLCD_LC_df_w_area, area_col, length_col = 'LENGTHKM', catchment_att = 'CAT'){
 
 #' @description This function takes the consolidated df from AOI_LC_w_area() and calculated the proportion share of each LC class at the PRMS segid scale
 #' @param NLCD_LC_df_w_area df output of AOI_LC_w_area that has LC data per COMID & PRMS id and area in (km2)
+#' @param area_col str col with catchment ('AREASQKM') or total upstream catchment area ('TOTDASQKM')
+#' @param length_col str name of col with length of catchment area ('LENGTHKM). Default 'LENGTHKM'
 #' @param @param catchment_att Specifies the catchment attribute to subset to. Must be either 'CAT', "ACC", 'TOT'. Default = "CAT"
-#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_LC_w_catchment_area)
+#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_NLCD_LC_w_catchment_area, area_col = 'AREASQKM')
 
   area_df <- NLCD_LC_df_w_area %>%
-    mutate(AREASQKM_approx = case_when(AREASQKM == 0 ~ LENGTHKM^2, TRUE  ~ AREASQKM)) %>% 
+    mutate(AREASQKM_approx = case_when(.data[[area_col]] == 0 ~ .data[[length_col]]^2, TRUE  ~ .data[[area_col]])) %>% 
```

Comment:
This approximation will not work for TOTDASQKM. I think we should only use an approximate area if area_col == AREASQKM

Did you check if there are any TOTDASQKM = 0?

Reply:
Yes, based on the source targets `p2_NLCD_LC_w_catchment_area`: 
42 comids have 0 TOTAL AREA. That amounts to 36 unique PRMS_segids. 

```
> p2_NLCD_LC_w_catchment_area %>% filter(TOTDASQKM == 0) %>% select(PRMS_segid, comid, AREASQKM, TOTDASQKM)
# A tibble: 42 × 4
   PRMS_segid comid   AREASQKM TOTDASQKM
   <chr>      <chr>      <dbl>     <dbl>
 1 1264_1     8077104        0         0
 2 1264_1     8077858        0         0
 3 1265_1     8078052        0         0
 4 141_1      2617690        0         0
 5 164_1      2743484        0         0
 6 17_1       1750665        0         0
 7 2129_1     9482534        0         0
 8 222_1      4152664        0         0
 9 238_1      4153808        0         0
10 240_1      4154548        0         0
# … with 32 more rows

```